src/tests/decider-distro-tests-rs/tests/test.rs

```diff
@@ -119,3 +533,1006 @@ async fn test_decider_installed() {
+        for step in 0..5 {
+            println!("step {step}");
+            let (method, params) = recv_request.recv().await.unwrap();
+            println!("\n\n{method}\n\n");
+            let response = match method.as_str() {
+                "eth_blockNumber" => {
+                    json!(to_hex(BLOCK_INIT))
+                }
+                "eth_getLogs" => {
+                    let log = serde_json::from_value::<LogsReq>(params[0].clone()).unwrap();
+                    Value::Array(vec![TestApp::log_test_app1(
+                        DEAL_ID_1,
+                        BLOCK_NUMBER_1,
+                        log.topics[1].as_str(),
+                    )])
+                }
+                "eth_sendRawTransaction" => {
+                    json!("0x55bfec4a4400ca0b09e075e2b517041cd78b10021c51726cb73bcba52213fa05")
+                }
+                "eth_getTransactionCount" => json!("0x1"),
+                "eth_gasPrice" => json!("0x3b9aca07"),
+                _ => panic!("unexpected method: {}", method),
+            };
+            send_response.send(Ok(response)).unwrap();
+        }
+    }
+    wait_decider(&mut client).await;
+    let deals = get_joined_deals(&mut client).await;
+    assert!(!deals.is_empty(), "decider didn't join any deal");
+    // The second run
+    update_config(&mut client, &oneshot_config()).await.unwrap();
+    {
+        // Reqs: blockNumber, getLogs, gasPrice, getTransactionCount and sendRawTransaction and getLogs for the old deal
+        for step in 0..6 {
+            println!("step {step}");
+            let (method, params) = recv_request.recv().await.unwrap();
+            println!("\n\n{method}\n\n");
+            let response = match method.as_str() {
+                "eth_blockNumber" => {
+                    json!(to_hex(BLOCK_NUMBER_2))
+                }
+                "eth_getLogs" => {
+                    let log = serde_json::from_value::<LogsReq>(params[0].clone()).unwrap();
+                    if step == 1 {
+                        json!([TestApp::log_test_app2(
+                            DEAL_ID_2,
+                            BLOCK_NUMBER_2,
+                            log.topics[1].as_str(),
+                        )])
+                    } else if step == 5 {
+                        json!([])
+                    } else {
+                        panic!("call eth_getLogs on the wrong step {step}");
+                    }
+                }
+                "eth_sendRawTransaction" => {
+                    json!("0x55bfec4a4400ca0b09e075e2b517041cd78b10021c51726cb73bcba52213fa05")
+                }
+                "eth_getTransactionCount" => json!("0x1"),
+                "eth_gasPrice" => json!("0x3b9aca07"),
+                _ => panic!("unexpected method: {}", method),
+            };
+            send_response.send(Ok(response)).unwrap();
+        }
+    }
+    wait_decider(&mut client).await;
+
+    let (last_seen, deals, mut workers) = {
```

Comment:
Maybe DRY this block?

Reply:
I wonder what this block should be called. I can't see how to unite this concepts